pmacApp/src/pmacTrajectory.cpp

```diff
@@ -294,6 +294,25 @@ asynStatus pmacTrajectory::getUserMode(int index, int *user) {
   return status;
 }
 
+asynStatus pmacTrajectory::getVelocityMode(int index, int *velocityMode) {
+  asynStatus status = asynSuccess;
+  static const char *functionName = "readVelocityMode";
+
+  debug(DEBUG_TRACE, functionName, "Called with index", index);
+
+  // Check the index is valid
+  if (index < 0 || index >= noOfValidPoints_) {
+    debug(DEBUG_ERROR, functionName, "Invalid index requested", index);
+    status = asynError;
+  }
```

Comment:
This index checking code block is used multiple times in this file.  Might be worth making it a function

Reply:
Also making the axis checking into a function.

Reply:
Done in [9dbb012](https://github.com/DiamondLightSource/pmac/pull/124/commits/9dbb012d0c76056907989cc63d24ca97e3250711)